apps/competitions-api/src/routes/inscriptions/createInscription.ts

```diff
@@ -1,11 +1,24 @@
 import { Router } from 'express';
 import { parseRequest, AuthenticatedRequest, checkAdminRole, checkRole } from '@competition-manager/utils';
-import { Competition$, BaseInscriptionWithRelationId$, BaseAdmin$ } from '@competition-manager/schemas';
+import { Competition$, BaseInscriptionWithRelationId$, BaseAdmin$, Athlete, Athlete$ } from '@competition-manager/schemas';
 import { z } from 'zod';
 import { prisma } from '@competition-manager/prisma';
 
 export const router = Router();
 
+const findAthleteWithLicense = async (license: number, oneDayAthletes: Athlete[] = []) => {
+    const athlete = oneDayAthletes.find((a) => a.license === license);
+    if (athlete) {
+        return athlete;
+    }
+    return await prisma.athlete.findFirstOrThrow({
+        where: {
+            license: license,
+            competitionId: null
+        }
+    });
+}
```

Comment:
Idée d'opti
```suggestion
    return oneDayAthletes.find((a) => a.license === license) || await prisma.athlete.findFirstOrThrow({
        where: {
            license: license,
            competitionId: null
        }
    })
}
```
mais je sais pas si avec le await ça marche sinon peut-être
```suggestion
    return await (oneDayAthletes.find((a) => a.license === license) || prisma.athlete.findFirstOrThrow({
        where: {
            license: license,
            competitionId: null
        }
    }))
}
```